### PR TITLE
OdsAboutModule - step 2 - Add menu

### PR DESCRIPTION
**Pull request by paulinea**

### CODE REVIEW

File: app/src/main/java/com/orange/ods/app/ui/modules/about/AboutCustomizationScreen.kt [hunk 1]
```diff
@@ -17,22 +17,16 @@ import androidx.compose.foundation.layout.FlowRow
...
+import com.orange.ods.app.ui.components.utilities.ComponentCountRow
 import com.orange.ods.app.ui.modules.Module
 import com.orange.ods.app.ui.modules.ModuleDetailColumn
 import com.orange.ods.compose.component.chip.OdsFilterChip
 import com.orange.ods.compose.text.OdsTextBody2
-import com.orange.ods.module.about.configuration.OdsAboutModuleConfiguration
-import com.orange.ods.module.about.configuration.OdsAboutShareData
+
 
 enum class AboutOptions(@StringRes val labelResId: Int) {
```
Comment (florentmaitre) [inline app/src/main/java/com/orange/ods/app/ui/modules/about/AboutCustomizationScreen.kt:31]: Should we rename `AboutOptions` to `AboutCustomizationOptions`?

File: app/src/main/java/com/orange/ods/app/ui/modules/about/AboutCustomizationScreen.kt [hunk 2]
```diff
@@ -41,53 +35,47 @@ enum class AboutOptions(@StringRes val labelResId: Int) {
...
+
 @OptIn(ExperimentalLayoutApi::class)
 @Composable
-fun AboutCustomizationScreen(navigateToAboutModule: () -> Unit, configureAboutModule: (OdsAboutModuleConfiguration) -> Unit) {
-    val context = LocalContext.current
-    ModuleDetailColumn(module = Module.About, onViewDemoButtonClick = { navigateToAboutModule() }) {
-        var selectedOptions by rememberSaveable { mutableStateOf(emptyList<AboutOptions>()) }
+fun AboutCustomizationScreen(navigateToAboutModule: () -> Unit, viewModel: AboutCustomizationViewModel) {
+    with(viewModel) {
+        ModuleDetailColumn(Module.About, onViewDemoButtonClick = { navigateToAboutModule() }) {
```
Comment (florentmaitre) [inline app/src/main/java/com/orange/ods/app/ui/modules/about/AboutCustomizationScreen.kt:45]:
```suggestion
        ModuleDetailColumn(module = Module.About, onViewDemoButtonClick = { navigateToAboutModule() }) {
```

File: app/src/main/java/com/orange/ods/app/ui/modules/about/AboutCustomizationContent.kt
Comment (florentmaitre) [inline app/src/main/java/com/orange/ods/app/ui/modules/about/AboutCustomizationContent.kt:1]: The content in this file is unused.

File: lib/src/main/java/com/orange/ods/compose/module/emptyview/OdsEmptyView.kt [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+import com.orange.ods.compose.component.content.OdsComponentContent
+import com.orange.ods.compose.component.content.OdsComponentImage
+import com.orange.ods.compose.component.utilities.BasicPreviewParameterProvider
+import com.orange.ods.compose.component.utilities.Preview
+import com.orange.ods.compose.component.utilities.UiModePreviews
+import com.orange.ods.compose.theme.OdsTheme
+
+
+@Composable
+fun OdsEmptyView(
```
[the remark this answers is not in the document]
Comment (paulinea) [reply lib/src/main/java/com/orange/ods/compose/module/emptyview/OdsEmptyView.kt:41]: It is public cause it is used by the about module which is not a part of the library module.

File: lib/src/main/java/com/orange/ods/compose/module/emptyview/OdsEmptyView.kt [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+     */
+    constructor(@RawRes animationRes: Int) : super(animationRes, "")
+
+    /**
+     * Creates an instance of [OdsEmptyViewIllustration].
+     *
+     * @param painter The painter to draw.
+     * @param contentScale The rule to apply to scale the image in this [OdsEmptyViewIllustration], [ContentScale.Crop] by default.
+     */
+    constructor(painter: Painter, contentScale: ContentScale = ContentScale.Crop) : super(
```
[the remark this answers is not in the document]
Comment (paulinea) [reply lib/src/main/java/com/orange/ods/compose/module/emptyview/OdsEmptyView.kt:114]: Yes you're right and I think this is not always the case. Maybe we would have to fix some signatures...

File: module-about/src/main/java/com/orange/ods/module/about/configuration/OdsAboutMenuItem.kt [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        painterResource(id = R.drawable.ic_dataprotection),
+        stringResource(id = R.string.ods_about_menu_privacy_policy),
+        100,
+        "about_privacy_policy",
+        FileAboutItem.FileFormat.Html
+    ),
+    FileAboutItem(
+        painterResource(id = R.drawable.ic_tasklist),
+        stringResource(id = R.string.ods_about_menu_terms_of_service),
+        101,
```
[the remark this answers is not in the document]
Comment (paulinea) [reply module-about/src/main/java/com/orange/ods/module/about/configuration/OdsAboutMenuItem.kt:34]: Yes, the need is to have grouped mandatory items but not necessary at the end or at the beginning of the menu.

File: module-about/src/main/java/com/orange/ods/module/about/configuration/OdsAboutMenuItem.kt [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+@Stable
+internal fun mandatoryMenuItems(): List<OdsAboutMenuItem> = listOf(
+    FileAboutItem(
+        painterResource(id = R.drawable.ic_dataprotection),
+        stringResource(id = R.string.ods_about_menu_privacy_policy),
+        100,
+        "about_privacy_policy",
+        FileAboutItem.FileFormat.Html
+    ),
+    FileAboutItem(
```
[the remark this answers is not in the document]
Comment (paulinea) [reply module-about/src/main/java/com/orange/ods/module/about/configuration/OdsAboutMenuItem.kt:31]: As discussed together, I moved it in an OdsAbout object to use nested classes as types.

File: module-about/src/main/java/com/orange/ods/module/about/configuration/OdsAboutMenuItem.kt [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+ * @property secondaryText The secondary text of the menu item displayed below the primary text.
+ */
+@Immutable
+sealed class OdsAboutMenuItem(val icon: Painter, val text: String, val positionIndex: Int, val secondaryText: String? = null)
+
+class FileAboutItem(
+    icon: Painter,
+    title: String,
+    positionIndex: Int,
+    val fileName: String,
```
[the remark this answers is not in the document]
Comment (paulinea) [reply module-about/src/main/java/com/orange/ods/module/about/configuration/OdsAboutMenuItem.kt:55]: I cannot set a default value for files with a `@RawRes Int` cause the file is not provided in the about module so it doesn't compile.
Another solution is to set mandatory parameters in the about module configuration for these files. What do you think?

File: module-about/src/main/java/com/orange/ods/module/about/navigation/OdsAboutNavigation.kt [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+import androidx.navigation.NavOptions
+import androidx.navigation.NavType
+import androidx.navigation.compose.composable
+import androidx.navigation.navArgument
+import androidx.navigation.navigation
+import com.orange.ods.module.about.OdsAboutHomeScreen
+import com.orange.ods.module.about.OdsAboutViewModel
+
+fun NavController.navigateToOdsAbout(navOptions: NavOptions? = null) {
+    navigate(OdsAboutDestinations.HomeRoute, navOptions)
```
[the remark this answers is not in the document]
Comment (paulinea) [reply module-about/src/main/java/com/orange/ods/module/about/navigation/OdsAboutNavigation.kt:27]: You're right, this is a mistake. We should use `OdsAboutDestinations.AboutRoute` here.

File: gradle.properties [hunk 9]
```diff
@@ -29,4 +29,5 @@ android.enableJetifier=true
 # Kotlin code style for this project: "official" or "obsolete":
 kotlin.code.style=official
 kapt.incremental.apt=true
+kapt.use.jvm.ir=false
```
[the remark this answers is not in the document]
Comment (paulinea) [reply gradle.properties:32]: I added this line cause I had an error on build and I found this workaround. This is not the case anymore so I remove it.

**Review by florentmaitre — APPROVED** [review]

**Pull request merged**